template.tpl

```diff
@@ -2134,6 +2323,41 @@ switch (data.eventType) {
     parameters = {filter: filter};
 
     break;
+    
+  case 'trackConsent':
+    const required = [
+      'consentScopes',
+      'basisForProcessing',
+      'consentUrl',
+      'consentVersion',
+      'domainsApplied',
+     ];
+    
+    const missing = required.filter(r => !data.hasOwnProperty(r));
```

Comment:
We can probably trust the UI to provide the properties once all marked non-empty; but probably also good to check they're not `undefined` since Variables can be used.